renderer/FrameBuffer.cpp

```diff
@@ -21,765 +21,311 @@ Project: The Dark Mod (http://www.thedarkmod.com/)
 }
 
-// switch from fbo to default framebuffer, copy content
-void LeavePrimary() {
-	if ( !primaryOn ) 
-		return;
-	primaryOn = false;
-	GL_CheckErrors();
-
-	GL_Viewport( 0, 0, glConfig.vidWidth, glConfig.vidHeight );
-	GL_Scissor( 0, 0, glConfig.vidWidth, glConfig.vidHeight );
-
-	if ( r_multiSamples.GetInteger() > 1 ) {
-		FB_ResolveMultisampling( GL_COLOR_BUFFER_BIT );
-		qglBindFramebuffer( GL_READ_FRAMEBUFFER, fboResolve );
+void FrameBuffer::AddRenderTexture( idImage *texture, GLenum attachment, int mipLevel ) {
+	if (texture->uploadWidth >> mipLevel != width || texture->uploadHeight >> mipLevel != height) {
+		common->Warning("Adding texture %s to framebuffer %s: size mismatch", texture->imgName.c_str(), name.c_str());
 	}
-	qglBindFramebuffer( GL_DRAW_FRAMEBUFFER, 0 );
-	qglBlitFramebuffer( 0, 0, globalImages->currentRenderImage->uploadWidth,
-	                    globalImages->currentRenderImage->uploadHeight,
-	                    0, 0, glConfig.vidWidth, glConfig.vidHeight, GL_COLOR_BUFFER_BIT, GL_LINEAR );
-
-	if ( r_showFBO.GetInteger() ) {
-		if ( r_multiSamples.GetInteger() > 1 ) {
-			FB_ResolveMultisampling( GL_DEPTH_BUFFER_BIT | GL_STENCIL_BUFFER_BIT );
-		}
-		qglBindFramebuffer( GL_FRAMEBUFFER, 0 );
-
-		GL_SetProjection( mat4_identity.ToFloatPtr() );
+	Bind();
+	qglFramebufferTexture2D(GL_FRAMEBUFFER, attachment, GL_TEXTURE_2D, texture->texnum, mipLevel);
+}
 
-		GL_State( GLS_DEFAULT );
-		qglDisable( GL_DEPTH_TEST );
-
-		programManager->oldStageShader->Activate();
-		Uniforms::Global* transformUniforms = programManager->oldStageShader->GetUniformGroup<Uniforms::Global>();
-		idMat4 ninety = mat4_identity * .9f;
-		ninety[3][3] = 1;
-		transformUniforms->modelViewMatrix.Set( ninety );
-		//transformUniforms->textureMatrix.Set( ninety );
-
-		GL_SelectTexture( 0 );
-		switch ( r_showFBO.GetInteger() ) {
-		case 1:
-			globalImages->shadowAtlas->Bind();
-			break;
-		case 2:
-			globalImages->currentDepthImage->Bind();
-			break;
-		case 3:
-			globalImages->shadowDepthFbo->Bind();
-			qglTexParameteri( GL_TEXTURE_2D, GL_DEPTH_STENCIL_TEXTURE_MODE, GL_DEPTH_COMPONENT );
-			break;
-		case 4:
-			ambientOcclusion->ShowSSAO();
-			break;
-		case 5:
-			bloom->BindBloomTexture();
-			break;
-		default:
-			globalImages->currentRenderImage->Bind();
-		}
-		RB_DrawFullScreenQuad();
-		transformUniforms->modelViewMatrix.Set( mat4_identity );
-		programManager->oldStageShader->Deactivate();
 
-		qglEnable( GL_DEPTH_TEST );
-	}
-	qglBindFramebuffer( GL_FRAMEBUFFER, 0 );
+renderCrop_t ShadowAtlasPages[42];
+idCVar r_fboResolution( "r_fboResolution", "1", CVAR_RENDERER | CVAR_FLOAT | CVAR_ARCHIVE, "internal rendering resolution factor" );
 
-	if ( r_frontBuffer.GetBool() ) {
-		qglFinish();
-	}
-	GL_CheckErrors();
+#if defined(_MSC_VER) && _MSC_VER >= 1800 && !defined(DEBUG)
+//#pragma optimize("t", off) // duzenko: used in release to enforce breakpoints in inlineable code. Please do not remove

```

Comment:
Maybe ask Duzenko once again and remove this dirty thing?

Reply:
Given that there's not really any code left in that file to which this could apply, I just deleted it. (And if it's needed again, it's in the history, anyway.)